File: q2/q2.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def addTwoNumbers(self, l1: ListNode, l2: ListNode) -> ListNode:
        carry = 0
        solution = ListNode(0)
        solution_next = solution
        while(l1 or l2 or carry ):
            carry, sum = divmod (( l1.val if l1 else 0) + (l2.val if l2 else 0) + carry, 10)
            
            solution_next.next = ListNode(sum)
            solution_next = solution_next.next
            l1 = l1.next if l1 else None
            l2 = l2.next if l2 else None
        
        return solution.next
        

File: q1-q9/q2/q2.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode() : val(0), next(nullptr) {}
 *     ListNode(int x) : val(x), next(nullptr) {}
 *     ListNode(int x, ListNode *next) : val(x), next(next) {}
 * };
 */
class Solution {
public:
ListNode* addTwoNumbers(ListNode* l1, ListNode* l2) {
      ListNode start(0);
      ListNode *start_next = &start;
      
      int transfer = 0;
      while(l1 || l2 || transfer) {
    
        int res = (l1? l1->val : 0) + (l2 ? l2->val : 0) + transfer;
        transfer = res / 10;
        
        start_next->next = new ListNode(res % 10);
        start_next = start_next->next;

        l1 = l1 ? l1->next : l1;
        l2 = l2 ? l2->next : l2;
      }
        
      return start.next;
    }
};
